substrate/frame/revive/src/lib.rs

```diff
@@ -1054,6 +1101,7 @@ where
 		code: Code,
 		data: Vec<u8>,
 		salt: Option<[u8; 32]>,
+		is_eth_call: bool,
```

Comment:
Sorry to be that guy but I think we should be strict about public functions with many arguments. What I don't like about this:
1) It describes where the call is coming from but not what it does. Naming it after what it does makes it much easier to understand: `bump_nonce`.

2) Let's avoid new booleans on such functions either. Please `enum BumpNonce`. Just makes it so much easier to write on the call site given that rust analyzer is still really viable on this repo.